streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamKTableJoinProcessor.java

```diff
@@ -58,22 +58,23 @@ public void init(final ProcessorContext context) {
 
     @Override
     public void process(final K1 key, final V1 value) {
-        // we do join iff keys are equal, thus, if key is null we cannot join and just ignore the record
-        // If {@code keyMapper} returns {@code null} it implies there is no match,
+        // We allow null keys unless {@code keyMapper} returns {@code null} and we ignore it as invalid.
```

Comment:
I guess we don't care about the original `key` any longer and only consider if `keyMapper` returns `null` or not?

Reply:
I think that makes sense - but that will mean removing the previous logic of "`mappedKey` is null implies key not found in global table"? (Original line 62)

Reply:
Well, if `mappedKey` is null then there can't be a match in the global table since we can't do a lookup with a null key. I think what @mjsax means here (correct me if wrong) is just that we could phrase it a bit differently to say something like
"If the mappedKey is null, we ignore it as invalid. This should never happen for KTables since keyMapper just returns the key, but for GlobalKTables a non-null key can result in a null mappedKey. Since there can't be a match for a null mappedKey, we drop it"

...or something. Thoughts?

Reply:
Yep makes sense. Have updated it now. Just noting here that this means we're changing a [test](https://github.com/apache/kafka/pull/9186/files#diff-e3715715832b244da2d8787362b0c570R230) we previously had